internal/controller/managedzone_controller_test.go

```diff
@@ -100,5 +109,56 @@ var _ = Describe("ManagedZoneReconciler", func() {
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(ContainSubstring("spec.domainName in body should match"))
 		})
+
+		It("status should move from false to true when provider error resolved", func() {
+			var p provider.Provider
+			p = fake.Provider{
+				EnsureManagedZoneFunc: func(mz *v1alpha1.ManagedZone) (provider.ManagedZoneOutput, error) {
+					return provider.ManagedZoneOutput{}, fmt.Errorf("provider error")
+				},
+			}
+			SetDNSProvider(p)
+			mz := &v1alpha1.ManagedZone{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "status-check",
+					Namespace: defaultNS,
+				},
+				Spec: v1alpha1.ManagedZoneSpec{
+					ID:         "something",
+					DomainName: "example.com",
+				},
+			}
+			err := k8sClient.Create(ctx, mz)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				err = k8sClient.Get(ctx, client.ObjectKey{Namespace: mz.Namespace, Name: mz.Name}, mz)
+				Expect(err).NotTo(HaveOccurred())
+				if meta.IsStatusConditionTrue(mz.Status.Conditions, string(v1alpha1.ConditionTypeReady)) {
+					return fmt.Errorf("expected the ready condition to be false")
+				}
+				if mz.Status.NameServers != nil {
+					return fmt.Errorf("expected nameservers not to be set on provider error")
+				}
+				if mz.Status.RecordCount != 0 {
+					return fmt.Errorf("expected record count to be 0")
+				}
+				return nil
```

Comment:
You might want to consider using gomega assertions here to be more consistent with all the other tests:

```.go
Eventually(func(g Gomega) {
	err = k8sClient.Get(ctx, client.ObjectKey{Namespace: mz.Namespace, Name: mz.Name}, mz)
	g.Expect(err).NotTo(HaveOccurred())
	g.Expect(mz.Status.Conditions).To(
		ContainElement(MatchFields(IgnoreExtras, Fields{
			"Type":               Equal(string(v1alpha1.ConditionTypeReady)),
			"Status":             Equal(metav1.ConditionTrue),
			"Reason":             Equal("Whatever this is meant to be"),
			"Message":            Equal("Whatever this is meant to be"),
			"ObservedGeneration": Equal(mz.Generation),
		})),
	)
	g.Expect(mz.Finalizers).To(ContainElement(ManagedZoneFinalizer))
	g.Expect(mz.Status.NameServers).To(BeNil())
	g.Expect(mz.Status.RecordCount).To(BeZero())
}, TestTimeoutMedium, time.Second).Should(Succeed())
```